Doc/library/hmac.rst

```diff
@@ -31,6 +31,21 @@ This module implements the HMAC algorithm as described by :rfc:`2104`.
       MD5 as implicit default digest for *digestmod* is deprecated.
 
 
+.. function:: digest(key, msg, digestmod)
+
+   Return digest of *msg* for given secret *key* and *digestmod*. The
+   function is equivalent to ``HMAC(key, msg, digestmod).digest()``, but
+   uses an optimized C or inline implementation, which is faster for messages
+   that fit into memory. The parameters *key*, *msg*, and *digestmod* have
```

Comment:
I realize `digestmod` matches the parameter name in hmac.new, but it is an antiquated name from the days when the only two hash functions were top level modules.  how about just calling it `digest` on the new API?

Are there _any_ hash function modules with a `new` method left anywhere that anyone actually uses?  Why continue to support that old idiom in new APIs?

Reply:
I though about it but then decided to keep the new function compatible to the old PEP.

Since you also dislike the old name, let's use ``digest`` in the new function.